libopae/src/sysfs.c

```diff
@@ -55,6 +55,11 @@ fpga_result sysfs_read_int(const char *path, int *i)
 	char buf[SYSFS_PATH_MAX];
 	int b;
 
+	if (path == NULL) {
```

Comment:
I'm on the fence about these new path checks for NULL.  On the one hand, it is a good thing to do this type of checking.  On the other, these are internal functions, and we have a lot of empirical evidence from our testing over time that we are not passing NULL to these. @r-rojo, do you have a preference here?

Reply:
open segfaults if user pass NULL, added null check before passing open

Reply:
Yes, I saw `open` segfault with NULL input parameter. I think @anandaravuri uncovered with the negative tests.